kong/runloop/balancer.lua

```diff
@@ -526,27 +532,38 @@ do
 
       upstreams_dict[up.name] = up.id
     end
-    return upstreams_dict
+    upstream_version = new_upstream_version
+    all_upstreams = upstreams_dict
+    return true
   end
-  _load_upstreams_dict_into_memory = load_upstreams_dict_into_memory
-
-
-  local opts = { neg_ttl = 10 }
+  _reset_all_upstreams = reset_all_upstreams
 
+  reload_all_upstreams = function()
+    if upstream_version == new_upstream_version then
+      return true
+    end
+    return reset_all_upstreams()
+  end
+end
 
-  ------------------------------------------------------------------------------
-  -- Implements a simple dictionary with all upstream-ids indexed
-  -- by their name.
-  -- @return The upstreams dictionary (a map with upstream names as string keys
-  -- and upstream entity tables as values), or nil+error
-  get_all_upstreams = function()
-    local upstreams_dict, err = singletons.cache:get("balancer:upstreams", opts,
```

Comment:
If this cache use is removed, Kong will have to repeat the upstreams update on every worker, which would be wasteful. I believe there's no need to remove this. Also, as `"balancer:targets:" .. upstream.id` and `"balancer:upstreams:" .. upstream_id` are still used, it increases the risk of inconsistency.